web-api/src/storage/postgres.rs

```diff
@@ -1263,3 +1272,60 @@ impl storage::Size for Storage {
         Ok(size)
     }
 }
+
+#[async_trait(?Send)]
+impl storage::IndexedProperties for Storage {
+    async fn load_schema(&self) -> Result<IndexedPropertiesSchema, Error> {
+        let mut tx = self.postgres.begin().await?;
+        let schema = self.load_schema_with_transaction(&mut tx).await?;
+        tx.commit().await?;
+        Ok(schema)
+    }
+
+    async fn extend_schema(
+        &self,
+        update: IndexedPropertiesSchemaUpdate,
+        max_properties: usize,
+    ) -> Result<IndexedPropertiesSchema, Error> {
+        let mut tx = self.postgres.begin().await?;
+        let mut schema = self.load_schema_with_transaction(&mut tx).await?;
+        schema.update(&update, max_properties)?;
+        self.extend_postgres_schema(&mut tx, &update).await?;
+        //TODO self.elastic.extend_elasticsearch_mapping(&update).await?;
+        tx.commit().await?;
+        Ok(schema)
+    }
+}
+
+impl Storage {
+    async fn load_schema_with_transaction(
+        &self,
+        tx: &mut Transaction<'_, Postgres>,
+    ) -> Result<IndexedPropertiesSchema, Error> {
```

Comment:
:art: in the other `Database` impls we usually take the `tx` instead of the `self` and also don't name the `_with_transaction` suffix as it should be clear from the function signature. it's probably good to align this as well.
```suggestion
    async fn load_schema(
        tx: &mut Transaction<'_, Postgres>,
    ) -> Result<IndexedPropertiesSchema, Error> {
```